.gitlab-ci.yml

```diff
@@ -266,18 +272,39 @@ TF_Tooling::Web::test:
     - cd ${env:TF_TFW_PATH}
     - npm i
     - npm run test
+    - npm run test:coverage
 
   artifacts:
     when: always
     name: "TestResults"
     paths:
     - ${env:TF_TFW_PATH}/junit.xml
+    - ${env:TF_TFW_PATH}/coverage/
     reports:
       junit: ${env:TF_TFW_PATH}/junit.xml
+      coverage_report:
+        coverage_format: cobertura
+        path: ${env:TF_TFW_PATH}/coverage/cobertura-coverage.xml
+
+  coverage: '/All files[^|]*\|[^|]*\s+([\d\.]+)/'
   tags:
     - unity
     - win10
 
+pages:
+  stage: deploy
+  dependencies:
+    - TF_Tooling::Web::test
+  script:
+    - mkdir .public
+    - cp ${env:TF_TFW_PATH}/coverage/* .public
+    - mv .public public
+  artifacts:
+    paths:
+      - public
+  only:
+    - main
```

Comment:
Why is this `only: main` when TouchFree doesn't have a `main` branch? Shouldn't this be `develop`? Not seeing this be run by GL in the latest pipeline for the branch.

Reply:
Oh yeah my thinking was "coverage for releases" but we don't have main to track that. I'll change it to develop.